controller/instance_manager_controller.go

```diff
@@ -1005,45 +1055,45 @@ func (imc *InstanceManagerController) canDeleteInstanceManagerPDB(im *longhorn.I
 			replica.Spec.NodeID == im.Spec.NodeID
 
 		if !hasPDBOnAnotherNode && !isUnusedReplicaOnCurrentNode {
-			return false, nil
+			return false, "no pdb on another node", nil
```

Comment:
I see the following logs when there is a blocking replica:

```
[longhorn-manager-mxdb9 longhorn-manager] time="2024-08-08T19:35:38Z" level=info msg="Node instance-manager-699da83c0e9d22726e667344227e096b is marked unschedulable but removing eweber-v126-worker-9c1451b4-kgxdq PDB is blocked: no pdb on another node " func="controller.(*InstanceManagerController).syncInstanceManagerPDB" file="instance_manager_controller.go:823" controller=longhorn-instance-manager node=eweber-v126-worker-9c1451b4-kgxdq
```

But it is possible to know which replica is blocking:

```
eweber@laptop:~/longhorn> kl get replica -oyaml | grep -i -e "-r-" -e "nodeID: eweber-v126-worker-9c1451b4-kgxdq" -e desireState -e currentState
    name: pvc-a73367a8-94d8-4d57-aaa1-da1365b723fb-r-fddc6af4
    desireState: running
    nodeID: eweber-v126-worker-9c1451b4-kgxdq
    currentState: running
...
```

This is a place we can augment the log with additional info as well. And in particular, it is a place that will help the Harvester debugging. Sorry to have missed it before!

TBH, I think it is fine to just include the name of the single `targetReplica` we are considering here in a format that you like. There is probably no need to rework the logic so that we have the whole list at once.

Reply:
Reasonable and handy to add the name of replica to the return messge, update.